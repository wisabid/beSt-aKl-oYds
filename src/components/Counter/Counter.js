import React from 'react';
import PropTypes from 'prop-types';

const Counter = ({betslipdata, subscriptions}) => {
    return (
        <div className="betslip">
            <span className="anchorl" title="Bet Slip">{betslipdata.length}</span>
            <span className="anchorl" title="Subscriptions">{subscriptions.length}</span>
        </div>
    )
}

Counter.propTypes = {
    betslipdata : PropTypes.array.isRequired,
    subscriptions : PropTypes.array.isRequired
}

export default Counter;