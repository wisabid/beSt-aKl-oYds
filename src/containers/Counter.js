import { connect } from 'react-redux';
import CounterComp from '../components/Counter';

const mapStateToProps = (state) => {
    return {        
        betslipdata : state.rL.betslipdata,
        subscriptions: state.rL.subscriptions
    }
}

const mapdispatchToProps = (dispatch) => {
    return {
        
    }
}

const Counter = connect(mapStateToProps, mapdispatchToProps)(CounterComp);

export default Counter;