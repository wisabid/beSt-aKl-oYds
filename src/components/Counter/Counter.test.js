import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Counter from './Counter';

describe('<Counter />', () => {
  const counterComp = shallow(<Counter betslipdata={[]} subscriptions={[]}/>);
  
  it('<Counter /> renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Counter betslipdata={[]} subscriptions={[]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('<Counter /> component matches the snapshot', () => {
    expect(counterComp).toMatchSnapshot();
  })

  it('<Counter /> displays a count of outcomes added to betslip', () => {
    expect(counterComp.find('[title="Bet Slip"]')).toHaveLength(1);
  });

  it('<Counter /> displays a count of subscriptions', () => {
    expect(counterComp.find('[title="Subscriptions"]')).toHaveLength(1);
  });

});
