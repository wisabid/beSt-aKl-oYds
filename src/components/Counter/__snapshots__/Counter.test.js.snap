// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> <Counter /> component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    betslipdata={Array []}
    subscriptions={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="anchorl"
          title="Bet Slip"
        >
          0
        </span>,
        <span
          className="anchorl"
          title="Subscriptions"
        >
          0
        </span>,
      ],
      "className": "betslip",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "anchorl",
          "title": "Bet Slip",
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "anchorl",
          "title": "Subscriptions",
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="anchorl"
            title="Bet Slip"
          >
            0
          </span>,
          <span
            className="anchorl"
            title="Subscriptions"
          >
            0
          </span>,
        ],
        "className": "betslip",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "anchorl",
            "title": "Bet Slip",
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "anchorl",
            "title": "Subscriptions",
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
